hpx/util/annotated_function.hpp

```diff
@@ -114,7 +114,8 @@ namespace hpx { namespace util
 #endif
 
     ///////////////////////////////////////////////////////////////////////////
-    namespace detail
+    /// \brief Gets useful information for the input function.
```

Comment:
I think we can remove this one. `namespace detail` is implicitly the "internal implementation details"-namespace.